### ISSUE TITLE
UWP Release Build requires Dependancy Microsoft.NET.CoreFramework.Debug.2.2.appx

### ISSUE THREAD

**Issue by cdm22** [microsoft/dotnet#1078]
Visual Studio Professional 2017    15.9.6
Building a **Release** Version of the APP.   
Why does it now add the **Debug** dependency? 

Using Package Microsoft.NETCore.UniversalWindowsPlatform  6.1.9   
   Dependencies x64: 
        Microsoft.NET.CoreRuntime.2.1.appx
        Microsoft.VCLibs.x64.14.00.appx

Upgrade to:
Microsoft.NETCore.UniversalWindowsPlatform 6.2.8
   Dependencies x64: 
        **Microsoft.NET.CoreFramework.Debug.2.2.appx**
        Microsoft.NET.CoreRuntime.2.1.appx
        Microsoft.VCLibs.x64.14.00.appx

**Comment by danielleiszen** [comment]
I have the same problem during install, if I uncheck "Compile with .NET Native tool chain" on project properties.

**Comment by cubed-it** [comment]
This additional dependency with almost 10MB per plattfrom is not particularly a lightweight. So what is it for?

**Comment by jtorjo** [comment]
Any work on this? 
This is still happening on latest 6.2 version - I simply can't release an app because of this!

**Comment by jtorjo** [comment]
Apparently, MS doesn't quite give a flying f*ck.
I'ts been over 6 MONTHS, and there's no resolution to this. 

What I do suggest is this: here's how to create an .appinstaller file that allows people to install your app. Not trivial at all, but I for one think it's the only thing that works
https://docs.microsoft.com/answers/questions/4027/uwp-cant-install-signed-application-non-ms-storeco.html

**Comment by seankanderson** [comment]
Creating the appinstaller file did not help me.  I got it to install the app but if I try to link to the file it just pulls up the XML in the user's browser and they don't know how to deal with that. Total non-solution when trying to distribute to beta testers.  What kills me is that this WAS working without issue.  I was testing this appx on clean Windows 10 VMs and it was installing beautifully.

**Comment by jtorjo** [comment]
@seankanderson I've answered you on the other forum (https://docs.microsoft.com/answers/questions/4027/uwp-cant-install-signed-application-non-ms-storeco.html) Please let me know if it works

**Comment by seankanderson** [comment]
I resolved this by reducing the AppxBundlePlatforms property down to only x64. 
The appxbundle with the missing dependencies was built/packaged with all of the of the platforms selected. 
This, for some reason, results in the build not rolling all of the dependencies up into an app DLL. In my case, notice Nevitium.UWP.dll in the appx contents comparison picture below.

<img width="1417" alt="appxbundle_csproj_diff-bad-left-good right" src="https://user-images.githubusercontent.com/1112636/72658598-315d8b80-3981-11ea-9c05-b1f47b68833f.png">


<img width="1025" alt="appxbundle_csproj_diff-bad-left-good right-appx-contents" src="https://user-images.githubusercontent.com/1112636/72658604-4fc38700-3981-11ea-9daf-b6b3f40aa1ee.png">

**Comment by jtorjo** [comment]
> I resolved this by reducing the AppxBundlePlatforms property down to only x64.
> The appxbundle with the missing dependencies was built/packaged with all of the of the platforms selected.
> This, for some reason, results in the build not rolling all of the dependencies up into an app DLL. In my case, notice Nevitium.UWP.dll in the appx contents comparison picture below.

Awesome to hear! Yeah, I think I missed to mention that I was only building for x64 as well (notice that the Package dependencies are only x64 related). Building for all platforms (x86, x64, arm) is probably a lot more complicated, but I assume doable.

On my side, I really only need x64.

Best,
John

**Comment by simader** [comment]
Any news on this?

When I try to install my UWP Release App I get on some x86 devices:
App installation failed with error message: Deployment of package Microsoft.NET.CoreRuntime.2.2_2.2.27902.3_x86__8wekyb3d8bbwe was blocked because the provided package has the same identity as an already-installed package but the contents are different. Increment the version number of the package to be installed, or remove the old package for every user on the system before installing this package. (0x80073cfb)

**Comment by PrinceCharles23** [comment]
almost to the day one year ago this was posted and there is no actionable reply from microsoft on the issue - great!

**Comment by jtorjo** [comment]
@PrinceCharles23 Don't expect MS to do anything about. Ever. I've simply just learned to live with it. Found workarounds - that's all you can do.

**Comment by PrinceCharles23** [comment]
> @PrinceCharles23 Don't expect MS to do anything about. Ever. I've simply just learned to live with it. Found workarounds - that's all you can do.

i am working with microsoft technologies for over 20 years now. so this is not news to me. makes me question my life decisions each time, tho.

anyways: do you have a workaround since you suggested that as a solution?

**Comment by simader** [comment]
What is your workaround?

For my case it seems that the unsigned package Microsoft.NET.CoreRuntime.2.2_2.2.27902.3_x86__8wekyb3d8bbwe is curropt, and i have to uninstall it on every PC where i want to install my UWP app.

**Comment by jtorjo** [comment]
You need to create an .appinstaller file, kinda' like shown here:

https://docs.microsoft.com/en-us/answers/questions/4027/uwp-cant-install-signed-application-non-ms-storeco.html
(note: for some reason, the link in there is broken)

The installer should look more or less like this
https://phot-awe.com/resources/test.appinstaller

**Comment by PrinceCharles23** [comment]
ok

> You need to create an .appinstaller file, kinda' like shown here:
> 
> https://docs.microsoft.com/en-us/answers/questions/4027/uwp-cant-install-signed-application-non-ms-storeco.html
> (note: for some reason, the link in there is broken)
> 
> The installer should look more or less like this
> https://phot-awe.com/resources/test.appinstaller

ok, thx! still have no clue what to do with this installer-file (and/or what i will have to add to make my app work) but - as i understand it - this just adds the culprit to the package but does not remove the neccessity to have the dependency in the first place. in other words: it just installs the debug dependency on production!?!

**Comment by jtorjo** [comment]
> ok, thx! still have no clue what to do with this installer-file (and/or what i will have to add to make my app work) but - as i understand it - this just adds the culprit to the package but does not remove the neccessity to have the dependency in the first place. in other words: it just installs the debug dependency on production!?!

Yes, that is entirely correct. I am not aware of anything else that truly works

**Comment by EhsanJahoon** [comment]
Seems it is related to Microsoft.NETCore.UniversalWindowsPlatform version !
The issue appears in 6.2.10 but it is works fine in 6.2.9

**Comment by jtorjo** [comment]
@EhsanJahoon No it doesn't . I've been using 6.2.9 for months and it's still the same. I think the issue dates back to 6.2.1 or something

**Comment by gsorensen** [comment]
I can confirm that whilst using 6.2.10 I also encountered this problem, and downgrading to 6.2.9 doesn't work either. Downgrading to the latest 6.1 release solves the issue in my case (The dependency is gone from the release build)

**Comment by devdeer-alex** [comment]
I want to add some strange stuff here: I have several releases named like "ReleaseTest", "ReleaseProd" and so on. All of them are without the DEBUG flag. If I deploy using the original "Release"-config it will not come with this error. It seams like MS is parsing for a magic string "Release" here and does generate this f....ing dependency whenever something is named differently.

**Comment by jtorjo** [comment]
@devdeer-alex That is very interesting, I will test this at some point.

**Comment by keithnicholson** [comment]
@jtorjo Any movement on this.  It is affecting me as well on more than just the Debug dependency.  It appears that an update, I'm guessing, to some other application has the same signature but different hash codes.

Microsoft.NET.CoreRuntime.2.2_2.2.27902.3_x64__8wekyb3d8bbwe was blocked because the provided package has the same identity
     | as an already-installed package but the contents are different

**Comment by Ruddy2007** [comment]
Keeping this issue alive... Any news? Also having the same issue.

**Comment by RemusVasii** [comment]
I do not know if this is the right way to solve the "Debug" thing when building a release version, but what worked for me was to edit the files "runtime.win10-??????.Microsoft.Net.UWPCoreRuntimeSdk.targets" found in "C:\Users\{CurrentUser}\.nuget\packages\...".

**Comment by RemusVasii** [comment]
It seems that using the option "Compile with .NET Native tool chain" is more or less a requirement for a UWP app these days, so using the hack I mentioned previously to edit the "targets" files will not really create a usable app, even if the dependency problem is "solved". In the end, I had to restructure my application (and remove some valuable functionality from it) in order to be able to compile it with .NET Native tool chain.

This is a little bit sad, because mine is a multiplatform app and the limitations imposed by compiling with .NET Native tool chain only exist for the UWP project. On all the other platforms (iOS, Android, macOS) the application works fine with dependencies, and there is no need to limit any app functionality.

**Comment by FrancyAngy** [comment]
the file you need is in `C:\Program Files (x86)\Microsoft SDKs\UWPNuGetPackages\runtime.win10-x64.microsoft.net.uwpcoreruntimesdk\2.2.14\tools\Appx`, but if you don't find it, i attached a version of both files found
[files.zip](https://github.com/microsoft/dotnet/files/12267815/files.zip)
